[Master] Add docs for HYPR authenticator (#3821)
wso2/docs-is: en/docs/guides/passwordless/hypr.md

# Passwordless login with HYPR

[HYPR](https://www.hypr.com/) is a passwordless authentication method that allows users to log in to applications using biometrics on their devices. As a FIDO-certified platform, HYPR eliminates risks prevalent in password-based authentication, such as phishing and password reuse.

This guide explains how you can use HYPR to add passwordless login to applications registered in the Identity Server.

## Prerequisites

- You need to configure the HYPR environment and have access to the HYPR control center. Learn how to do it in the [HYPR documentation](https://docs.hypr.com/hyprcloud/docs/cc-std).

> - You need to [set up the sample]({{base_path}}/guides/adaptive-auth/adaptive-auth-overview/#set-up-the-sample) application.

Reviewer: The use of a SAML application could lead to the perception that this connector is exclusively compatible with apps based on SAML protocols. Therefore, I suggest including a note to clarify that this connector is also compatible with applications based on OIDC protocols.

Author: Shall we add it like this :arrow_down: 

`You need to set up your application. For this tutorial, we will be using the [sample SAML application]({{base_path}}/guides/adaptive-auth/adaptive-auth-overview/#set-up-the-sample) application. Note that you can use any application you wish to use.`

4. Copy the `org.wso2.carbon.identity.application.authenticator.hypr.common-1.0.3.jar` file from the extracted folder and paste it to `<IS_HOME>/repository/components/lib`.

### Deploy the HYPR Rest API

To deploy the HYPR Rest API:

1. Copy the `api#hypr.war` file from the extracted artifacts folder and paste it to `<IS_HOME>/repository/deployment/server/webapps`.
2. Add the following configuration in the `<IS_HOME>/repository/conf/deployment.toml` file.

    ``` toml
    [[resource.access_control]]
    context = "(.*)/api/hypr/v1/authentication/status/(.*)"
    secure = "false"
    http_method = "GET"

    [tenant_context]
    enable_tenant_qualified_urls = "true"
    enable_tenanted_sessions = "true"
    rewrite.custom_webapps=["/api/hypr/"]
    ```

### Deploy the HYPR login page

To deploy the HYPR login page from the extracted artifacts folder, copy the `hyprlogin.jsp` and paste it to `<IS_HOME>/repository/deployment/server/webapps/authenticationendpoint` folder.

## Configure HYPR as a federated authenticator

> Let's configure WSO2 IS with HYPR as a federated authenticator.

Reviewer: ```suggestion
To configure WSO2 IS with HYPR as a federated authenticator.
```

1. On the Management console, go to **Main** > **Identity** > **Identity Providers** > **Add**.
2. Enter an **Identity Provider Name**, **Display Name**, and **Description**.
4. Expand the **Federated Authenticators** > **HYPR Configuration** and fill in the following details:

    | Field name    | Description   |
    |---------------|---------------|

>     | Enable        | Select this to enable HYPR authenticator for the login flow of the application    |

Reviewer: ```suggestion
    | Enable        | Select this to enable HYPR authenticator for the identity provider.    |
```

>     | Default       | Select this if the HYPR authenticator should be the default authenticator for the application    |

Reviewer: ```suggestion
    | Default       | Select this if the HYPR authenticator should be the default authenticator for the identity provider.    |
```

    | Base URL      | This refers to the base URL you received from HYPR upon creating a tenant for your organization.  |
    | Relying Party App ID  | This refers to the App ID you received for the application you created in the HYPR Control Center.    |

>     | API Token     | This refers to a new API token that is specifically generated for the HYPR App via the Control Center.    |

Reviewer: ```suggestion
    | API Token     | This refers to the application's API access token that is generated in the HYPR Control Center.    |
```

    ![HYPR configurations]({{base_path}}/assets/img/guides/hypr-configurations.png)

6. Click **Update** to save the configurations.

> ## Add HYPR authenticator to the app

Reviewer: ```suggestion
## Enable HYPR login for the application
```
